indico/modules/search/result_schemas.py

```diff
@@ -199,6 +199,29 @@ class EventNoteResultSchema(ResultSchemaBase):
     event_path = fields.Method('_get_event_path', dump_only=True)
 
     def _get_url(self, data):
+        from indico.modules.events.sessions.models.blocks import SessionBlock
+        from indico.util.string import slugify
+
+        ev = Event.get(data['event_id'])
+
+        if not ev:
+            return self._get_fallback_url(data)
+
+        # notes attached to the meeting itself don't get fragments
+        if ev.note and ev.note.id == data['note_id']:
+            return url_for('events.display', event_id=data['event_id']) + f"?note={data['note_id']}"
+        else:
+            # find the parent block
+            blocks = SessionBlock.query.filter(SessionBlock.session.has(event=ev, is_deleted=False))
```

Comment:
Can I do something like this?
`SessionBlock.query.filter(SessionBlock.session.note.has(id=3))`
instead of looping over all blocks?

Reply:
something like this should work: `SessionBlock.query.filter(SessionBlock.session.has(Session.note.has(id=3)))`

Reply:
Thanks :)